packages/client/src/devices/MicrophoneManager.ts

```diff
@@ -139,8 +139,10 @@ export class MicrophoneManager extends InputMediaDeviceManager<MicrophoneManager
         },
       );
 
-      this.noiseCancellationRegistration = this.registerFilter(
-        noiseCancellation.toFilter(),
+      const { ready: noiseCancellationRegistration, unregister } =
+        this.registerFilter(noiseCancellation.toFilter());
+      this.noiseCancellationRegistration = noiseCancellationRegistration.then(
+        () => unregister,
```

Comment:
I had to change things around the noise cancellation filter a bit, but I haven't looked to deep into it. Probably the code can be simplified in this file.